src/main/java/com/plivo/api/models/message/MessageLister.java

```diff
@@ -63,6 +72,29 @@ public MessageLister errorCode(final Long errorCode) {
     return this;
   }
 
+  @Override  
+  /**
+  * Actually list instances of the resource.
+  */
+  public ListResponse<Message> list() throws IOException, PlivoRestException {
+    validate();
+    Response<ListResponse<Message>> response = obtainCall().execute();
+
+    handleResponse(response);
+    
+    try 
+    {
+      Field meta = response.body().getClass().getDeclaredField("meta");
+      meta.setAccessible(true);
+      meta.set(response.body(), new MessageMeta(response.body().getMeta()));
```

Comment:
can u cross verify the meta.set ? are u replacing response.body() with getMeta()?

Reply:
I did test this. It is working as expected.

No, response.body() isn't getting replaced. response.body() is a ListResponse object which has a "meta" field. The value of that field is being replaced by a new MessageMeta object